service/matching/matchingEngine.go

```diff
@@ -105,6 +105,7 @@ func NewEngine(taskManager persistence.TaskManager,
 		}),
 		metricsClient: metricsClient,
 		config:        config,
+		queryTaskMap:  make(map[string]chan *workflow.RespondQueryTaskCompletedRequest),
```

Comment:
Should this be a buffered channel.

Reply:
It should not matter. Because the receiving side should always be there. If the receiving side is not there (because of timeout), the channel will be removed from the map. But make it buffered with size of 1 will make it more safe that the sending side will never block.